Reading user "do not disturb" status

Mazuh opened the pull request:
Is there any icon that should be used for this "snooze"? That's how it looks currently:
![screen shot 2018-08-11 at 19 21 36](https://user-images.githubusercontent.com/6237270/43996570-dbda9996-9d9b-11e8-98cc-acdfdd8eaf45.png)

haskellcamargo commented:
Can you take a look on this, @duynguyenhoang?

duynguyenhoang commented:
Let's me check it

duynguyenhoang commented on sclack/components.py:
```diff
@@ -564,13 +564,32 @@ def keypress(self, size, key):
+        else:
+            return ''
+
+    def change_snooze_indicator(self, is_snoozed=None):
```
I think `is_snoozed=False`, am I correct?

duynguyenhoang replied:
Also change status indicator icon when is_snoozed is changed

Mazuh replied:
> I think `is_snoozed=False`, am I correct?

I meant to set a snooze to `True` or `False` only if someone explicitly said so. It was just a semantic idea. But yeah, it could be only `is_snooze=False` (and then I'd remove the `if` below).

> Also change status indicator icon when is_snoozed is changed

That led me to the other thread's question: should this snooze be a separated indicator or just be the same as `presence_icon`?

duynguyenhoang replied:
I think this indicator should be only one, but we have some statuses
* Snooze
* Online

duynguyenhoang replied:
I tried to update it slightly, please check https://i.imgur.com/at0XuwE.png

duynguyenhoang commented on sclack/components.py:
```diff
@@ -564,13 +564,32 @@ def keypress(self, size, key):
+
         if is_online:
-            presence_icon = ('presence_active', ' {} '.format(get_icon('online')))
+            self.presence_icon = ('presence_active', ' {} '.format(get_icon('online')))
```
Here, we should add snooze icon like:
```
if is_snoozed:
            self.presence_icon = ('presence_active', ' {} '.format(get_icon('snooze')))
        
```
I have checked https://nerdfonts.com/ and see we can use `"snooze": "\uF9B1",`

Mazuh replied:
so the presence icon should indicate snooze too? 🤔

duynguyenhoang replied:
yes, it should be snooze icon.

Mazuh replied:
ok, then, I'll do it this evening

Mazuh commented:
@duynguyenhoang I updated the code applying your reviews. Also removed the (snoozed) string as it seems not needed anymore, and the display string for snooze is set as "presence_snooze". All good with that?

I have a problem, however. I don't know what triggered this, but my icon have a black background:
![screen shot 2018-08-15 at 08 53 41](https://user-images.githubusercontent.com/6237270/44147441-910cb45e-a06a-11e8-8bb1-3b61978a8c70.png)

duynguyenhoang commented on sclack/components.py:
```diff
@@ -564,13 +564,26 @@ def keypress(self, size, key):
+    @property
+    def body(self):
+        if self.is_snoozed:
+            presence_icon = ('presence_snooze', ' {} '.format(get_icon('snooze')))
```
Could you please change `presence_snooze` to `presence_active`, It is actually the color of the icon
So the icon becomes
![snooze](https://i.imgur.com/at0XuwE.png)

Mazuh replied:
Aaaah, got it, thanks. I'll do.

Should this 'snoozed' string be there too?

duynguyenhoang replied:
I think yes, good to have, because not everyone use the icon so the text should be helpful

Mazuh commented:
done! 👍 

![screen shot 2018-08-15 at 11 10 11](https://user-images.githubusercontent.com/6237270/44152497-1171d1a4-a07c-11e8-86d8-0366e0c47349.png)

Mazuh commented:
Also, I saw a _Todo_ at https://github.com/haskellcamargo/sclack/pull/64 that's related to this snooze feature. I left the `is_snoozed` in the `state` object so you can use it there.

duynguyenhoang commented:
Please rebase your branch, I am going to merge it then.
Thank you for your efffort.

Mazuh commented:
@duynguyenhoang did it!

duynguyenhoang commented:
I mean rebase your branch, merge your commits into one commit.
Like below command

```
git rebase -i HEAD~5
```

Mazuh commented:
Oh, I thought you were going to "squash and merge" via GitHub, so I just used rebase to sync with `master`. Anyway, did it too.
